### ISSUE TITLE
CSharpRemoveUnnecessaryCastDiagnosticAnalyzer fails to consider langversion

### ISSUE THREAD

**Issue by Naine** [dotnet/roslyn#50000]
**Version Used**: 3.8.0-5.20604.10 (9ed4b774d20940880de8df1ca8b07508aa01c8cd), Visual Studio 16.8.3

Consider this code:

```csharp
    ushort Foo(string? s)
        => s is null ? (ushort)1234 : ushort.Parse(s);
```

In C#9, with target-typed conditional expressions, this code will compile without the cast, because the subexpressions in the conditional each have a type target of `ushort`, which 1234 is a valid literal for.

Prior to C#9, if the cast is removed, the type of the literal is `int`, so the conditional expression has a type of `int` and this fails to compile because `int` is not implicitly convertible to `ushort`.

In dev16.8, the IDE0004 analyzer identifies the cast as unnecessary and offers to remove it, regardless of the language version used. If the language version is lower than 9.0, the IDE0004 diagnostic is a false-positive, and applying the code-fixer to remove the cast introduces a CS8370 error (or CS0266 under an older compiler).
